### PR TITLE
Add Chainlink oracle; Add TRU to poolValue

### PR DESCRIPTION
**Pull request by vanruch**
Chainlink address 0xa7becdd46648110112c85dd489a70f1119c81698

### CODE REVIEW

**Review by barrutko — APPROVED** [review]

**Comment by barrutko** [comment]
Maybe let's remove the other oracle contract? The one we don't intend to use anymore.

**Review by hal909 — APPROVED** [review]

File: contracts/truefi/TruPriceChainlinkOracle.sol [hunk 1]
```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.10;
+
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+import {ITruPriceOracle} from "./interface/ITruPriceOracle.sol";
+import {IUniswapPair} from "./interface/IUniswapPair.sol";
```
Comment (BenjaminCosman) [inline contracts/truefi/TruPriceChainlinkOracle.sol:6]: This input looks unused?

File: contracts/truefi/TruPriceChainlinkOracle.sol [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    using SafeMath for uint256;
+
+    IChainLink public chainlinkOracle;
+
+    constructor(IChainLink _chainlinkOracle) public {
+        chainlinkOracle = _chainlinkOracle;
+    }
+
+    /**
+     * @dev converts from USD with 18 decimals to TRU with 8 decimals
```
Comment (BenjaminCosman) [inline contracts/truefi/TruPriceChainlinkOracle.sol:22]: From the comment I would expect to see a factor of 10^10 somewhere - how does dividing/multiplying by only 100 accomplish this conversion?

File: contracts/truefi/TrueFiPool.sol [hunk 3]
```diff
@@ -162,6 +179,14 @@ contract TrueFiPool is ITrueFiPool, ERC20, ReentrancyGuard, Ownable {
         _;
     }
 
+    /**
+     * @dev only lender can perform borrowing or repaying
```
Comment (BenjaminCosman) [inline contracts/truefi/TrueFiPool.sol:183]: This comment is wrong (copied from onlyLender)

File: contracts/truefi/TruPriceChainlinkOracle.sol [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+     */
+    function usdToTru(uint256 amount) external override view returns (uint256) {
+        return amount.div(safeUint(chainlinkOracle.latestAnswer())).div(100);
```
Comment (BenjaminCosman) [inline contracts/truefi/TruPriceChainlinkOracle.sol:25]: I see that `latestAnswer()` is deprecated in favor of `latestRoundData()`, though it also doesn't look like the distinction is important.

File: contracts/truefi/TruPriceChainlinkOracle.sol (hunk 2 above)
Comment (hal909) [reply contracts/truefi/TruPriceChainlinkOracle.sol:22]: add comment to explain the math here

File: contracts/truefi/TrueFiPool.sol [hunk 5]
```diff
@@ -243,7 +296,7 @@ contract TrueFiPool is ITrueFiPool, ERC20, ReentrancyGuard, Ownable {
      * @return pool value in TUSD
      */
     function poolValue() public view returns (uint256) {
-        return liquidValue().add(loansValue());
+        return liquidValue().add(loansValue()).add(truValue());
```
Comment (hal909) [inline contracts/truefi/TrueFiPool.sol:299]: consider removing truValue() function

File: contracts/truefi/TruPriceChainlinkOracle.sol (hunk 1 above)
Comment (hal909) [reply contracts/truefi/TruPriceChainlinkOracle.sol:6]: resolved

File: contracts/truefi/TruPriceChainlinkOracle.sol (hunk 2 above)
Comment (hal909) [reply contracts/truefi/TruPriceChainlinkOracle.sol:22]: resolved

File: contracts/truefi/TruPriceChainlinkOracle.sol (hunk 4 above)
Comment (hal909) [reply contracts/truefi/TruPriceChainlinkOracle.sol:25]: not sure if this matters, so marking as resolved

**Pull request merged**